python/cuspatial/cuspatial/core/geodataframe.py

```diff
@@ -168,7 +168,9 @@ def _recombine_columns(
 
         The output is meant for GeoDataFrame._from_data.
         """
-        if not geo_columns.index.equals(data_columns.index):
+        if not (
+            geo_columns.empty or data_columns.empty
+        ) and not geo_columns.index.equals(data_columns.index):
```

Comment:
De Morgan's rules can clean this up.

```suggestion
        if not (
            geo_columns.empty or
            data_columns.empty or
            geo_columns.index.equals(data_columns.index)
        ):
```

Reply:
Thanks! Will commit